pointers/pointers.c

```diff
@@ -28,25 +34,85 @@ void string_copy(char *x, char *y)
 */
 char *find_char(char *str, int c)
 {
+    // int sLen = string_length(str);
 
+    // for (int i = 0; i <= sLen; i++)
+    // {
+    //     if (str[i] == c)
+    //     {
+    //         return &str[i];
+    //     }
+    // }
+
+    // Bit more concise. Doesn't need a full for loop implementation, since we can just use the null terminator to stop the loop at worst. No pesky worrying about off by 1 errors.
+    while (*str != 0)
+    {
+        if (*str == c)
+        {
+            return str;
+        }
+        str++;
+    }
+
+    // In case `c` doesn't exist in `str`, otherwise we'd have an edge case where the app crashes due to not returning at all
+    return NULL;
 }
 
 /*
     Searches the input string `haystack` for the first instance of
     the string `needle`. This function returns a pointer that points
     to the first instance of the string `needle` in the input
-    string `haystack`. 
+    string `haystack`.
 
     Do not use the `strstr` function from the standard library.
 */
 char *find_string(char *haystack, char *needle)
 {
+    int hLen = strlen(haystack);
+    int nLen = strlen(needle);
+    char *substring = malloc(nLen * sizeof(char));
+
+    while (*haystack != 0)
+    {
+        char *startChar = find_char(haystack, needle[0]);
+
+        if (*startChar == 0 || (&startChar + (nLen - 1)) > &haystack[hLen - 1])
```

Comment:
I'm a little confused - why are you comparing `&startChar` and `&haystack`?

Reply:
Trying (and probably mucking up) some pointer arithmetic to see if the string we're looking for would fit in the remaining length of haystack. Need to circle back to this function in particular, as it has a bit of refactoring churn that has probably introduced more issues than it fixed.

Reply:
aaah I see. ok, we'll leave this one